lib/pbench/server/api/resources/query_apis/datasets_update.py

```diff
@@ -107,7 +119,11 @@ def complete(
         """
         auditing: dict[str, Any] = context["auditing"]
         attributes = auditing["attributes"]
-        attributes["access"] = context["access"]
         if summary["failure"] == 0:
-            dataset.access = context["access"]
+            if context.get("access"):
+                attributes["access"] = context["access"]
+                dataset.access = context["access"]
+            if context.get("owner"):
+                attributes["owner"] = context["owner"]
+                dataset.owner_id = context["owner"]
```

Comment:
Since we don't care about the value at that point, line 123 should be
```
            if "access" in context:
```
Alternatively, you could write lines 123-125 as
```
            access = context.get("access")
            if access:
                attributes["access"] = access
                dataset.access = access
```
which costs an extra line, but saves a couple of look-ups.

The same applies to lines 126-128.